packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/Grid/GridItem.tsx

```diff
@@ -84,14 +84,11 @@ export function GridItem<Item extends ObjectItem = ObjectItem>(props: GridItemPr
   } = props
   const {t} = useTranslation()
 
-  const treeEditing = useTreeEditingEnabled()
-  const treeEditingDisabledByOption = parentSchemaType?.options?.treeEditing === false
-  const legacyEditing = treeEditingDisabledByOption || treeEditing.legacyEditing
+  const {enabled, legacyEditing} = useTreeEditingEnabled()
 
   // The edit portal should open if the item is open and:
   // - tree array editing is disabled
-  // - legacy array editing is enabled (e.g. in a Portable Text editor)
-  const openPortal = open && (!treeEditing.enabled || legacyEditing)
+  const openPortal = open && (!enabled || legacyEditing)
```

Comment:
Removing this since we don't want to provide this right now (or even in the future)